cluster/kubernetes/kubernetes.go

```diff
@@ -300,6 +304,32 @@ func (c *Cluster) getAllowedNamespaces() ([]apiv1.Namespace, error) {
 	return namespaces.Items, nil
 }
 
+func (c *Cluster) IsAllowedResource(id flux.ResourceID) bool {
+	if len(c.allowedNamespaces) == 0 {
+		// All resources are allowed when all namespaces are allowed
+		return true
+	}
+
+	namespace, kind, name := id.Components()
+	namespaceToCheck := namespace
+
+	if namespace == resource.ClusterScope {
```

Comment:
`ClusterScope` will only be the namespace if the ID is from parsing manifests (cluster/kubernetes/resource/resource.go) or via the sync code (cluster/kubernetes/sync.go). I don't think it can be supplied by a user (as in, you can't construct such a resource ID from user input).

At the minute this doesn't matter, since no operations in the API can be applied to cluster-scope resources (so there's nowhere a user would supply such a thing). But it does mean this check is brittle, if used on user input.

Reply:
Yep, I agree. However, I  am not sure how to make this better. 

Regarding the API, even if a user can somehow provide the ID of a cluster-scoped resource, I assume that we will be consistent in the representation of the ID.

Tangentially, I don't like the using a `<cluster>` namespace to indicate cluster-scoped. I think makes much more sense to use an empty string,  but I guess that's besides the point.

Reply:
> Tangentially, I don't like the using a `<cluster>` namespace to indicate cluster-scoped. I think makes much more sense to use an empty string, but I guess that's besides the point.

The trouble with the empty string is that it also means "not supplied". In some cases, "not supplied" lines up with "has no namespace" -- but in other cases, those are distinct.

Reply:
gotcha

Reply:
Let's go with this for now, and give ourselves some room to figure out how to improve it.